core/src/com/dev/flying/kiwi/gamecore/systems/BodyActorUpdateSystem.java

```diff
@@ -14,33 +16,39 @@
  * This system renders sprites at a position based on the world, for a given actor.
  * Created by Steven on 9/23/2015.
  */
-public class PhysicsActorRenderSystem extends IteratingSystem {
+public class BodyActorUpdateSystem extends IteratingSystem {
     private ComponentMapper<Box2DBodyComponent> physicsMapper;
     private ComponentMapper<ActorComponent> actorMapper;
 
-    private Array<Entity> renderQueue;
-    private Batch batch;
+    private Array<Entity> entitiesToUpdate;
 
-    public PhysicsActorRenderSystem(Batch batch) {
+    public BodyActorUpdateSystem() {
         super(Family.all(Box2DBodyComponent.class, ActorComponent.class).get());
         this.physicsMapper = ComponentMapper.getFor(Box2DBodyComponent.class);
         this.actorMapper = ComponentMapper.getFor(ActorComponent.class);
-        this.renderQueue = new Array<>();
+        this.entitiesToUpdate = new Array<>();
     }
 
     @Override
     protected void processEntity(Entity entity, float deltaTime) {
-        renderQueue.add(entity);
+        entitiesToUpdate.add(entity);
     }
 
-    public void drawRenderQueue() {
-        TextureRegion tr;
-//        for (Entity e : this.renderQueue) {
-//            tr = new TextureRegion(drawableMapper.get(e).texture, 0, 0, 64,64);
-//            batch.draw(tr, positionMapper.get(e).x, positionMapper.get(e).y, positionMapper.get(e).x, positionMapper.get(e).y, 32f, 32f, 1.0f, 1.0f, 10f);
-//            //batch.draw(drawableMapper.get(e).texture, positionMapper.get(e).x, positionMapper.get(e).y);
-//        }
-        renderQueue.clear();
+    public void drawRenderQueue(Batch batch) {
+        ShapeActor actor;
+        Body body;
+
+        for (Entity e : this.entitiesToUpdate) {
+            actor = actorMapper.get(e).actor;
+            body = physicsMapper.get(e).body;
+
+            actor.setPosition(body.getPosition().x, body.getPosition().y);
+            actor.setRotation(MathUtils.radiansToDegrees * body.getAngle());
+
+            actor.draw(batch, 1);
```

Comment:
This system shouldn't draw items, the stage should.
Issue raised https://github.com/NZKiwi32/RTS-Game/issues/10